src/pages/AddItem.js

```diff
@@ -14,24 +34,62 @@ export default function AddItem(props) {
     setPurchaseFrequency(e.target.value);
   };
 
-  const createListItem = (e) => {
-    e.preventDefault();
-    db.collection('generated_token')
-      .add({
-        item_name: itemName,
-        purchase_frequency: parseInt(purchaseFrequency),
-        last_purchased: lastPurchased,
-      })
-      .then((documentReference) => {
-        console.log('document reference ID', documentReference.id);
-      })
-      .catch((error) => {
-        console.log(error.message);
-      });
-    setItemName('');
-    setLastPurchased(null);
+  const checkForDuplicates = () => {
+    // we would like to potentially normalize white space but unsuccessfull so far
+    const normalizedUserInput = itemName
+      .toLowerCase()
+      .replace(/[^\w\s]|_/g, '');
+
+    const matchingItemName = listItems.filter((item) => {
+      const normalizedDataBaseItem = item.item_name
+        .toLowerCase()
+        .replace(/[^\w\s]|_/g, '');
+      console.log(normalizedDataBaseItem);
+      return normalizedDataBaseItem === normalizedUserInput;
+    });
+
+    if (matchingItemName.length > 0) {
+      return false;
+    } else {
+      return true;
+    }
   };
 
+  async function createListItem(e) {
+    e.preventDefault();
+    try {
+      const result = await checkForDuplicates(itemName);
+      if (result) {
+        db.collection('generated_token')
+          .add({
+            item_name: itemName,
+            purchase_frequency: parseInt(purchaseFrequency),
+            last_purchased: lastPurchased,
```

Comment:
I don't think it's related to the changes addressed in this PR, but I found a little bug happening with this `lastPurchased` variable. If the user adds the item name, and doesn't touch the dropdown to select anything, the state isn't altered, and it keeps the initial value of `null`.  The ways I can think addressing this might be initializing the variable with the value of 7 (since the option 'soon' is the one the user intended to select when they don't alter the dropdown) or adding an option tag asking the user to choose a frequency.  Here's a video showing what happened in case my explanation was too confusing 😅 


https://user-images.githubusercontent.com/20430736/115466173-158d0900-a206-11eb-9bc6-3aa92b787d58.mov

Reply:
Great catch, @apdsrocha! I like your idea of setting the default value to 7!

Reply:
@apdsrocha good eye! I totally agree that initializing the state to 7 also makes a lot of sense! Jessica or I can update that pretty easily!